### PR TITLE
Docs on payments and allocations behavior on unconfirmed order

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
Related to https://github.com/mirumee/saleor/pull/6829

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2678 Document allocations behavior with unconfirmed orders](https://app.clickup.com/t/2549495/SALEOR-2678)

File: docs/dashboard/orders.md [hunk 1]
```diff
@@ -14,6 +14,8 @@ There are essentially two order types in Saleor:
 
 - Draft orders - These are orders created by administrators, which have not yet been released to the system. For example, store operators can create orders during discussions with customers, then save them as drafts until approval is gained from management to send replacement goods. Draft orders can be edited in the same way that an order is created.
 
+- Unconfirmed orders - Orders placed by customers when order confirmation feature is turned on. The payment on such orders is being authorized when order is placed but captured later - when staff [confirms the order](#how-to-confirm-an-order).
```
Comment (maarcingebala) [inline docs/dashboard/orders.md:17]: "when order confirmation feature is turned on" - this part could link to the section where it is described how to turn on/off order confirmation.

File: docs/dashboard/orders.md [hunk 2]
```diff
@@ -136,9 +138,21 @@ When order confirmation is enabled, order needs to be confirmed by staff in orde
 
 1. Select an unconfirmed order from order list to open the order details
 
-2. Click _confirm&nbsp;order_ button on the bottom
+2. There is a possiblity to edit the order at this point. However there are a few things to be aware when doing so:
+
+- When order address gets changed, existing allocations remains untouched.
+- New allocations are created whenever staff increase order line amount or add a new order line.
+- Allocations will only get altered for variants with active inventory tracking set.
```
Comment (maarcingebala) [inline docs/dashboard/orders.md:145]: Here we could also link to the page when users can control tracking inventory for a variant.

File: docs/dashboard/orders.md [hunk 3]
```diff
@@ -136,9 +138,21 @@ When order confirmation is enabled, order needs to be confirmed by staff in orde
...
+- When order line is deleted, all existing allocations of that order line are deleted with it.
+- If staff tries to create / increase amount on an order line with greater amount of variant that is available, insufficient stock error will be displayed.
+- When manipulating order lines after order address gets changed, there might be a case that the items cannot be allocated for new address.
```
Comment (maarcingebala) [inline docs/dashboard/orders.md:148]: I would add one more sentence to describe why it can happen.

File: docs/dashboard/orders.md [hunk 4]
```diff
@@ -136,9 +138,21 @@ When order confirmation is enabled, order needs to be confirmed by staff in orde
...
+
+3. Click _confirm&nbsp;order_ button on the bottom
 
 After operation is finished, order status should be _unfulfilled_.
+Keep in mind that staff might change the order value before it's confirmed by manipulating the order lines. If such case happens, then after the confirmation, charged amount can be:
+- higher than order total - staff should perform miscellaneous refund for the amount that was overpaid.
+- lower than order total - staff should create a separate payment in order to clear the ballance.
```
Comment (maarcingebala) [inline docs/dashboard/orders.md:155]:
```suggestion
- lower than order total - staff should create a separate payment in order to clear the balance.
```

File: docs/dashboard/orders.md [hunk 5]
```diff
@@ -136,9 +138,21 @@ When order confirmation is enabled, order needs to be confirmed by staff in orde
...
+- Allocations will only get altered for variants with active inventory tracking set.
+- When order line is deleted, all existing allocations of that order line are deleted with it.
+- If staff tries to create / increase amount on an order line with greater amount of variant that is available, insufficient stock error will be displayed.
+- When manipulating order lines after order address gets changed, there might be a case that the items cannot be allocated for new address.
+
+3. Click _confirm&nbsp;order_ button on the bottom
 
 After operation is finished, order status should be _unfulfilled_.
+Keep in mind that staff might change the order value before it's confirmed by manipulating the order lines. If such case happens, then after the confirmation, charged amount can be:
+- higher than order total - staff should perform miscellaneous refund for the amount that was overpaid.
```
Comment (maarcingebala) [inline docs/dashboard/orders.md:154]: Here we could add a link to order refund docs.

File: docs/dashboard/orders.md (hunk 4 above)
Comment (maarcingebala) [reply docs/dashboard/orders.md:155]: "staff should create a separate payment" - if I understand correctly, it suggests that Saleor supports creating multiple payments for an order, but this is not possible. Staff users cannot create separate payments, if there is already a partial payment. I think in this case we want staff users to decide with the customer how to handle outstanding balance.

**Review by karolkielecki — APPROVED** [review]

**Pull request merged**